File: fe2-android/app/src/main/java/com/github/dedis/popstellar/SingleEvent.java
package com.github.dedis.popstellar;

public class SingleEvent<T> {

  private final T mContent;

  private boolean hasBeenHandled = false;

  public SingleEvent(T content) {
    if (content == null) {
      throw new IllegalArgumentException("null values not allowed in an Event");
    }
    mContent = content;
  }

  public T getContentIfNotHandled() {
    if (hasBeenHandled) {
      return null;
    } else {
      hasBeenHandled = true;
      return mContent;
    }
  }

  public boolean hasBeenHandled() {
    return hasBeenHandled;
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/SingleEvent.kt
package com.github.dedis.popstellar

class SingleEvent<T>(content: T?) {
    private val mContent: T
    private var hasBeenHandled = false

    init {
        requireNotNull(content) { "null values not allowed in an Event" }
        mContent = content
    }

    val contentIfNotHandled: T?
        get() = if (hasBeenHandled) {
            null
        } else {
            hasBeenHandled = true
            mContent
        }

    fun hasBeenHandled(): Boolean {
        return hasBeenHandled
    }
}